AbstractFactory.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace HelloWorld
+{
+    public abstract class HomeDevicesFactory
```

Comment:
The implementation is ok, but let's have a better look at the topic.

Quotation from the pattern definition:
_The Abstract Factory design pattern provides an interface for creating families of **related** or **dependent** objects dependent objects which must be used together.._

The example you've chosen does reflect the **relation** idea. 
I'd also like you to have a look on the other article with an example, where the objects are not only related (alike), but they interact between themselves:
https://www.dofactory.com/net/abstract-factory-design-pattern

Additional task:
try re-writing your example to introduce some dependency between your "devices", e.g. washing machine cleans dishes after the toast or a hot drink is made, hoover cleans crumbs after the toast is made.